pyaedt/modeler/cad/component_array.py

```diff
@@ -566,7 +589,80 @@ def lattice_vector(self):
         >>> oModule.GetLatticeVectors()
 
         """
-        return self.__app.omodelsetup.GetLatticeVectors()
+        lattice_vectors = self.__app.omodelsetup.GetLatticeVectors()
+
+        lattice_vectors = [float(vec) * AEDT_UNITS["Length"][self.__app.modeler.model_units] for vec in lattice_vectors]
+        return lattice_vectors
+
+    @pyaedt_function_handler()
+    def get_component_objects(self):
+        """Get 3D component center.
+
+        Returns
+        -------
+        dict
+            Dictionary of the center position and part name for all 3D components.
+
+        """
+        component_info = {}
+        component_names = self.component_names
+        for component in component_names.values():
+            parts = self.__app.modeler.user_defined_components[component].parts
+            for part_name in parts.values():
+                if component not in component_info:
+                    center = self.__app.modeler.user_defined_components[component].center
+                    scaled_center = [
+                        float(cen) * AEDT_UNITS["Length"][self.__app.modeler.model_units] for cen in center
+                    ]
+                    component_info[component] = [scaled_center, str(part_name)]
+                else:
+                    component_info[component].append(str(part_name))
+
+        return component_info
+
+    @pyaedt_function_handler()
+    def get_cell_position(self):
+        """Get cell position.
+
+        Returns
+        -------
+        dict
+            Dictionary of the center position and part name for all 3D components.
```

Comment:
Same description as before. Need update. Also is not a dict, it is a List of List. I would also add a description of the shape (it is basically a matrix and each element is a list itself).